webpack/assets/javascripts/react_app/API.js

```diff
@@ -30,9 +30,6 @@ export default {
       type: 'put',
       dataType: 'json',
       data: data,
-      success: function (response, textstatus, jqXHR) {
```

Comment:
This could be a good place to do the presentation update.

Now, the update is done optimistically when the ajax call is made (`NotificationActions.markAsRead`) which is probably OK, but then there should be provision to reverse if the ajax call returns an error.

Alternatively, the update could be made here after success confirmation is received.